util/json.go

```diff
@@ -77,12 +77,55 @@ func MustMarshalJSON(x interface{}) []byte {
 // Thereby, it is converting its argument to the representation expected by
 // rego.Input and inmem's Write operations. Works with both references and
 // values.
-func RoundTrip(x *interface{}) error {
+func RoundTrip(x *interface{}) (interface{}, error) {
 	bs, err := json.Marshal(x)
 	if err != nil {
-		return err
+		return nil, err
+	}
+	err = UnmarshalJSON(bs, x)
+	if err != nil {
+		return nil, err
+	}
+	return *x, err
+}
+
+// JSONify recursively converts x into native-JSON types.
+// Deepcopies x, and does not mutate the underlying structure.
+func JSONify(x *interface{}) (interface{}, error) {
```

Comment:
[nit] Let's not export that if possible. Once it's part of the API, it'll be promoted to more importance than it warrants 😅